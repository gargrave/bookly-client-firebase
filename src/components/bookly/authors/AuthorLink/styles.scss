.bookly-author-link {
  color: #6e6969;
  font-size: 1.1rem;
}
