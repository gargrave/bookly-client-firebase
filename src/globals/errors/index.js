// @flow
import { parseError, parseFbError } from './apiErrorParser';
import { validationErrors } from './validationErrors';

export {
  parseError,
  parseFbError,
  validationErrors,
};
