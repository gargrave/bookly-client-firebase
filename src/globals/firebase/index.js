import firebase from 'firebase';
import 'firebase/firestore';

import firebaseConfig from '../../secrets/firebaseConfig.js';

let initialized = false;
let db;

if (!initialized) {
  firebase.initializeApp(firebaseConfig);
  db = firebase.firestore();
  initialized = true;
}

function docToObject(doc) {
  return {
    id: doc.id,
    ...doc.data(),
  };
}

function timestamp() {
  return firebase.firestore.FieldValue.serverTimestamp();
}

export {
  db,
  docToObject,
  timestamp,
};
