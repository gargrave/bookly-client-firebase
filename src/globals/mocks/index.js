import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
