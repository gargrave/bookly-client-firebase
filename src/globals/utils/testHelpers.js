import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
