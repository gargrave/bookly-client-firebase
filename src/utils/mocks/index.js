import authorMocks from './data/authors';
import userMocks from './data/users';

export {
  authorMocks,
  userMocks,
};
