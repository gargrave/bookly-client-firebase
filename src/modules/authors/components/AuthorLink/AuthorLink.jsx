// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import { object } from 'prop-types';

import type { Author } from '../../flowtypes';

import { localUrls } from '../../../../globals/urls';

import styles from './AuthorLink.css';

type Props = {
  author: Author,
  history?: Object,
};

const AuthorLink = ({
  author,
}: Props) => {
  const authorId = author.id || '';
  return (
    <p className={styles.authorLink}>
      by&nbsp;
      <Link to={`${localUrls.authorsList}/${authorId}`}>
        {author.firstName} {author.lastName}
      </Link>
    </p>
  );
};

AuthorLink.propTypes = {
  author: object.isRequired,
  history: object,
};

export default AuthorLink;
