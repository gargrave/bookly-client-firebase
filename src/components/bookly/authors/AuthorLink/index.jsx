// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import { object } from 'prop-types';

import type { Author } from '../../../../constants/flowtypes/';

import { localUrls } from '../../../../constants/urls';
import { buildClasses } from '../../../../utils/cssHelpers';

import './styles.css';

type Props = {
  author: Author,
  history?: Object,
};

function AuthorLink({
  author,
}: Props) {
  return (
    <p className={buildClasses(['author-link'])}>
      by&nbsp;
      <Link to={`${localUrls.authorsList}/${author.id}`}>
        {author.firstName} {author.lastName}
      </Link>
    </p>
  );
}

AuthorLink.propTypes = {
  author: object.isRequired,
  history: object,
};

export default AuthorLink;
