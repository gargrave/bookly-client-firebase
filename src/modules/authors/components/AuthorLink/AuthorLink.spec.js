import { ComponentBuilder } from '../../../../utils/testHelpers';

import { authorMocks } from '../../../../globals/mocks';

import AuthorLink from './AuthorLink';

const defaultProps = {
  author: authorMocks[0],
};

const builder = new ComponentBuilder(
  AuthorLink, defaultProps,
);

describe('AuthorLink', () => {
  let component;

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    component = builder.shallowGetComponent();
    expect(component.find('.bookly-author-link')).toHaveLength(1);
  });
});
