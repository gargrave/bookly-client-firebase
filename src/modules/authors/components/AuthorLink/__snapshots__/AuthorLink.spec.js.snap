// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorLink matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorLink
    author={
      Object {
        "bookCount": 1,
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Aldous",
        "id": "oiw223h3qndlksadffl3ksdf",
        "lastName": "Huxley",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "by ",
        <Styled(Link)
          replace={false}
          to="/authors/oiw223h3qndlksadffl3ksdf"
        >
          Aldous
           
          Huxley
        </Styled(Link)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "by ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Aldous",
            " ",
            "Huxley",
          ],
          "replace": false,
          "to": "/authors/oiw223h3qndlksadffl3ksdf",
        },
        "ref": null,
        "rendered": Array [
          "Aldous",
          " ",
          "Huxley",
        ],
        "type": [Function],
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "by ",
          <Styled(Link)
            replace={false}
            to="/authors/oiw223h3qndlksadffl3ksdf"
          >
            Aldous
             
            Huxley
          </Styled(Link)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "by ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Aldous",
              " ",
              "Huxley",
            ],
            "replace": false,
            "to": "/authors/oiw223h3qndlksadffl3ksdf",
          },
          "ref": null,
          "rendered": Array [
            "Aldous",
            " ",
            "Huxley",
          ],
          "type": [Function],
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
